From 3c402cafee78b45a3f39e1b93347a090b2d40690 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 13 Apr 2023 19:58:49 +0200
Subject: [PATCH 5/9] Create cloud user if its missing and configure doas

Allow the init__create_default_user to be overridden in case we want
support other distros in the future.
---
 lib/tiny-cloud/init | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/lib/tiny-cloud/init b/lib/tiny-cloud/init
index 2294c28..9596889 100644
--- a/lib/tiny-cloud/init
+++ b/lib/tiny-cloud/init
@@ -14,6 +14,7 @@ INIT_ACTIONS_EARLY="
     expand_root
     install_hotplugs
     set_network_config
+    create_default_user
 "
 INIT_ACTIONS_MAIN="
     save_userdata
@@ -78,6 +79,22 @@ init__set_network_config() {
     assemble-interfaces
 }
 
+init__create_default_user() {
+    local user="$CLOUD_USER"
+    # dont do anythign if it already exists
+    if getent passwd "$user" >/dev/null; then
+        return
+    fi
+
+    addgroup "$user"
+    adduser -h "/home/$user" -s /bin/sh -G "$user" -D "$user"
+    addgroup "$user" wheel
+    echo "$user:*" | chpasswd -e
+
+    mkdir -p "$ROOT"/etc/doas.d
+    echo 'permit nopass :wheel' > "$ROOT/etc/doas.d/wheel.conf"
+}
+
 ### init-main functions
 
 init__set_hostname() {
-- 
2.40.1

